sharktank/tests/models/llama/prefill_tests.py

```diff
@@ -0,0 +1,149 @@
+import torch
+from sharktank.examples.paged_llm_v1 import *
+from sharktank.utils import tokenizer
+from sharktank.utils import hf_datasets
+import unittest
+from pathlib import Path
+
+
+class BaseLlamaTest(unittest.TestCase):
+    def setUp(self):
+        raise NotImplementedError("Subclasses should implement this method.")
+
+    def createConfigModel(self, kv_cache_type):
+        return LlamaModelConfig(
+            hp=configs.LlamaHParams.from_gguf_props(self.dataset.properties),
+            block_seq_stride=16,
+            kv_cache_type=kv_cache_type,
+            device=self.device,
+            activation_dtype=self.activation_dtype,
+            attention_dtype=self.activation_dtype,
+        )
+
+    def runPrefill(self, kv_cache_type):
+        config = self.createConfigModel(kv_cache_type)
+        model = PagedLlamaModelV1(self.dataset.root_theta, config)
+        generator = TorchGenerator(model, self.tokenizer_config)
+        batch = generator.begin_batch(self.prompts)
+        batch.prefill()
+
+        bs, *_ = batch.logits.shape
+        assert len(batch.seq_lens) == bs
+        greedy_token_logit = 0.0
+        for b, seq_len in enumerate(batch.seq_lens):
```

Comment:
Do we need a for loop here, if we only require the last token logit?

Reply:
Done